src/stcal/ramp_fitting/ramp_fit.py

```diff
@@ -263,8 +265,15 @@ def suppress_one_group_saturated_ramps(ramp_data):
         # the remainder of the ramp.
         sat_groups = np.zeros(intdq.shape, dtype=int)
         sat_groups[np.where(np.bitwise_and(intdq, sat_flag))] = 1
+        jump_groups = np.zeros(intdq.shape, dtype=int)
+        jump_groups[np.where(np.bitwise_and(intdq, jump_flag))] = 1
+        sat_groups[np.where(intdq == 1)] = 1
+        jump_groups[np.where(intdq == 4)] = 1
         nsat_groups = sat_groups.sum(axis=0)
-        wh_one = np.where(nsat_groups == (ngroups - 1))
+        njump_groups = jump_groups.sum(axis=0)
+        nbad_groups = nsat_groups + njump_groups
```

Comment:
It would be a little easier to read if the jump_groups statements are grouped together, and similarly for the sat_groups statements.